src/_guides/openapi/augmenting-generated-openapi.md

```diff
@@ -0,0 +1,349 @@
+          token: ${{secrets.BUMP_TOKEN}}
+          file: api/openapi.public.yaml # 👈 updated this
+```
+
+Now whenever anyone commits it'll publish the automatically created filtered `openapi.public.yaml` instead of the original version, and we won't even have to think about it again.
+
+You could also use this approach to deploy _two different_ sets of documentation, maybe on different [Bump.sh Hubs](https://bump.sh/api-catalog) for different teams with different permissions. 🙌
+
+**Another scenario:** what if we have been adding some new endpoints which aren't quite ready yet? Bump.sh users can use the `x-beta: true` [vendor extension](/help/specification-support/doc-beta/) and filter that just as you would with `x-internal`. Another approach would be to use tags to mark things as beta, because openapi-filter will let you use tags as flags with the `--checkTags` flag. Here we can look for the `Beta` tag, and filter out any operations that use it.
+
+```
+  '/webhook/orders':
+    post:
+      operationId: post-orders
+      summary: Save Shopify Order
+      tags:
+        - Webhooks
+        - Beta
+      requestBody:
+        content:
+          # snip
+```
+
+```
+$ openapi-filter -- --flags Beta --checkTags api/openapi.yaml api/openapi.public.yaml
+```
+
+That extra `--checkTags` means openapi-filter will look into the `tags` array, and remove any that have tags mentioned in flags! 
+
+For the Bump.sh GitHub Action that would look like this:
+
+```
+      - name: Filter beta endpoints
+        run: |
+          npx openapi-filter -- --flags Beta --checkTags api/openapi.yaml api/openapi.public.yaml
+
+      - name: Deploy API documentation
+        uses: bump-sh/github-action@v1
+        with:
+          doc: <DOC-ID>
+          token: ${{secrets.BUMP_TOKEN}}
+          file: api/openapi.public.yaml # 👈 updated this
+```
+
+Ok, enough with removing things. What about adding and changing things? 
+
+## Enriching OpenAPI with Descriptions & Examples 
+
+Product, marketing, or tech writers often focus more on the  "why" and "how" of the OpenAPI descriptions, helping prepare them to be used as API Documentation after developers have traditionally set up the skeleton based mainly on the "what". 
+
+For teams following an API Design First workflow (where you're managing your description documents as a source of truth and using them to power/test your code) then you probably don't have a problem, as your non-technical users can just modify those files, but teams who are following API Code-first have some problems to deal with here. When generating OpenAPI from source code with comments or annotations, or generated from HTTP traffic, any changes made to the generated OpenAPI will be lost next time they generate. How can that be avoided?
+
+OpenAPI has a new experimental extension called Overlays, put together by the OAI (OpenAPI Initiative), tooling vendors, and community members that all came together to make a specification. 
```

Comment:
Not sure it's important to mention “OpenAPI has a new...” here. The specification is led by the OAI, but it's not directly related to the OpenAPI specification. Which might confuse a bit the reader (I'm thinking about our AsyncAPI users :slightly_smiling_face:)

Reply:
Here I'm saying "OpenAPI has a" as in the "OpenAPI ecosystem/community/world has a", I've intentionally said that and not "The OpenAPI Specification has a", but either way it's definitely fair to say that its directly related to OpenAPI. 

Here's the introduction in the README:

> The Overlay specification defines a way of creating documents that contain information to be merged with an OpenAPI description at some later point in time, for the purpose of updating the OpenAPI description with additional information.

From the specification:

> The extends property can be used to indicate that the Overlay was designed to update a specific OpenAPI document. Where no extends is provided it is the responsibility of tooling to apply the Overlay documents to the appropriate OpenAPI document(s).

I'm sure AsyncAPI users can use it if they like but the article is about OpenAPI and getting into side quests might be too much?

Reply:
I was not going into a side quests. What I understood from the [Overlay specification](https://github.com/OAI/Overlay-Specification/blob/main/versions/1.0.0.md#actionObject) is that it says “to be applied to the **target document**” and all along the spec it talks about a “target document”. 

There's only two occurrences of “OpenAPI document” in the spec:
- In the _Traits example_ where it says “ a target document (such as an OpenAPI document) 
- In the definition of the `extends` field (which is optional).

Anyway, for generic documents I guess the [JSON Patch](https://www.rfc-editor.org/rfc/rfc6902.html) RFC is enough and we can consider the Overlay spec to be an OpenAPI specific tooling.